### PR TITLE
assessment: _canShowMarking & _canShowModelAnswer settings not applied on restore #2693

### PR DESCRIPTION
**Pull request by eleanor-heath**
https://github.com/adaptlearning/adapt_framework/issues/2693

### CODE REVIEW

**Comment by oliverfoster** [comment]
this needs some rework @eleanor-heath

File: js/adapt-assessmentArticleModel.js [hunk 1]
```diff
@@ -77,12 +77,31 @@ define([
...
+        component.set({
+          _canShowFeedback: assessmentConfig._questions._canShowFeedback,
+          _canShowMarking: assessmentConfig._questions._canShowMarking,
+          _canShowModelAnswer: assessmentConfig._questions._canShowModelAnswer
+        });
+      });
+
+      this.set('_currentQuestionComponentIds', this._currentQuestionComponents.map(function(comp) {
+        return comp.get('_id');
+      }));
```
Comment (oliverfoster) [inline js/adapt-assessmentArticleModel.js:102]:
```suggestion
```
Unnecessary / unused

File: js/adapt-assessmentArticleModel.js [hunk 2]
```diff
@@ -77,12 +77,31 @@ define([
...
+      this._currentQuestionComponents = this.findDescendantModels('components', { where: { _isQuestionType: true } });
+
+      var assessmentConfig = this.getConfig();
+      this._currentQuestionComponents.forEach((component) => {
+        component.set({
+          _canShowFeedback: assessmentConfig._questions._canShowFeedback,
+          _canShowMarking: assessmentConfig._questions._canShowMarking,
+          _canShowModelAnswer: assessmentConfig._questions._canShowModelAnswer
+        });
+      });
```
Comment (oliverfoster) [inline js/adapt-assessmentArticleModel.js:98]:
```suggestion
      const assessmentQuestionsConfig = this.getConfig()._questions;
      this._getAllQuestionComponents().forEach(component => {
        component.set({
          _canShowFeedback: assessmentQuestionsConfig._canShowFeedback,
          _canShowMarking: assessmentQuestionsConfig._canShowMarking,
          _canShowModelAnswer: assessmentQuestionsConfig._canShowModelAnswer
        });
      });
```

No need to store the array on the model instance.

**Review by oliverfoster — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by ethan-lp — APPROVED** [review]
👀

**Pull request merged**